engine/src/test/java/org/hibernate/search/test/backend/lucene/ScheduledCommitPolicyTest.java

```diff
@@ -80,6 +80,26 @@ public boolean evaluate() {
 	}
 
 	@Test
+	@BMRule(targetClass = "org.hibernate.search.backend.impl.lucene.IndexWriterHolder",
+			targetMethod = "commitIndexWriter()",
+			action = "throw new NullPointerException(\"Fake internal error\")",
+			name = "timerDisruptingError")
+	public void testErrorHandlingOnBackgroundThread() throws Exception {
+		writeData( sfAsyncExclusiveIndex, 2 );
+		final CountingErrorHandler errorHandler = (CountingErrorHandler) sfAsyncExclusiveIndex.getSearchFactory().getErrorHandler();
+		assertConditionMet( new Condition() {
+			@Override
+			public boolean evaluate() {
+				// It's going to commit once each millisecond, and produce a failure each time.
+				// So "4" is just a random number higher than 0, but high enough to
```

Comment:
Do you mean "2"?

Not critical, but maybe you can use a method or a constant do describe this better